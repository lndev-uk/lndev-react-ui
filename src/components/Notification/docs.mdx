import { Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Notification from './';

# Notification

Notifications are available for any length of text, as well as an optional dismiss button.
The example below are using the UI system variants.

<Preview>
  <Story id="components-notification--example" />
</Preview>
<Props of={Notification} />

## UX Variants

Notications can also be use in UX variants

<Preview>
  <Story id="components-notification--variants" />
</Preview>

## Dismissal

You Can also enable dismissal manual an automatic dismissal functionality.

<Preview>
  <Story id="components-notification--dismissal" />
</Preview>

## Stateless

For more advance implementation you can bypass the embeded state and control
the visibility from outside with the onClose callback

<Preview>
  <Story id="components-notification--stateless" />
</Preview>

## Animated

In the example above by using `dismissal=true and onClose and stateless=true` you can make the notificaiton persistant.
This means that you only need to use 1 notification per tree. Has long the notification is present using onClose
callback you can reset the notification by using an external state to which ever content and variant required/

You can also use Animations. For more info on animations please read.

<Preview>
  <Story id="components-notification--animated" />
</Preview>
