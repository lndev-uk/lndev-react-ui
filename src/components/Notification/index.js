//@flow

import React from 'react';
import PropTypes from 'prop-types';

type NotificationType = 'success' | 'info' | 'warning' | 'error';

export type NotificationProps = {
  type: NotificationType,
  children: React.ReactNode
};
// onClick a funciton if defined add close button
// onTimeout a function that if defined will use a timer to call the callback
export default function Notification(
  props: NotificationProps
): React.ReactNode {
  const { type, children } = props;
  const typeClassName = type ? 'notification--' + type : '';
  return <div className={`notification ${typeClassName}`}>{children}</div>;
}

Notification.propTypes = {
  type: PropTypes.oneOf(['success', 'info', 'warning', 'error']),
  children: PropTypes.node.isRequired
};
