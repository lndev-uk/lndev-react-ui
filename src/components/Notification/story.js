import React from 'react';
import Notification from '.';

export default {
  title: 'Notification'
};

export const Type = () => (
  <div>
    <div>
      <Notification type="success">
        This is a success notification!
      </Notification>
    </div>
    <div>
      <Notification type="info">This is a info notification!</Notification>
    </div>
    <div>
      <Notification type="warning">
        This is a warning notification!
      </Notification>
    </div>
    <div>
      <Notification type="error">This is a error notification!</Notification>
    </div>
  </div>
);
