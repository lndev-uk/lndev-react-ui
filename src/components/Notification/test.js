import React from 'react';
import { render } from '@testing-library/react';
import Notification from '.';

describe('Notification', () => {
  it('matches snapshot success', () => {
    const { container } = render(
      <Notification type="success">
        <p>This is a test</p>
      </Notification>
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot info', () => {
    const { container } = render(
      <Notification type="info">
        <p>This is a test</p>
      </Notification>
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot info', () => {
    const { container } = render(
      <Notification type="warning">
        <p>This is a test</p>
      </Notification>
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot info', () => {
    const { container } = render(
      <Notification type="error">
        <p>This is a test</p>
      </Notification>
    );
    expect(container).toMatchSnapshot();
  });

  it('matches children', () => {
    const { getByText } = render(
      <Notification type="error">
        <p>This is a test</p>
      </Notification>
    );
    const element = getByText('This is a test');
    expect(element.localName).toBe('p');
  });
});
