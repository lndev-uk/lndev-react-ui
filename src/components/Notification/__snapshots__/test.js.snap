// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification matches snapshot info 1`] = `
<div>
  <div
    class="notification notification--info"
  >
    <p>
      This is a test
    </p>
  </div>
</div>
`;

exports[`Notification matches snapshot info 2`] = `
<div>
  <div
    class="notification notification--warning"
  >
    <p>
      This is a test
    </p>
  </div>
</div>
`;

exports[`Notification matches snapshot info 3`] = `
<div>
  <div
    class="notification notification--error"
  >
    <p>
      This is a test
    </p>
  </div>
</div>
`;

exports[`Notification matches snapshot success 1`] = `
<div>
  <div
    class="notification notification--success"
  >
    <p>
      This is a test
    </p>
  </div>
</div>
`;
